observer/src/observer/cli.py

```
@@ -0,0 +1,291 @@
+@main.command()
+def register() -> None:
+    """Register a Claude Code session (called by SessionStart hook)."""
+    from observer.services.registration import (
+        HookInput,
+        ensure_daemon_running,
+        register_session,
+    )
+
+    raw = sys.stdin.read()
+    if not raw.strip():
+        raise SystemExit("No input received on stdin.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)